src/PlayerResults.re

```diff
@@ -3,119 +3,171 @@ open LeaderboardUtils;
 open PlayerStatsUtils;
 open Streaks;
 open Types;
-open Queries;
-open EloUtils;
-open UseCommunitySettings;
-open ApolloHooks;
+
+module Query = [%relay.query
+  {|
+    query PlayerResultsQuery($communityName: String!, $playerName: String!) {
+      results_connection(
+        where: {
+          _and: [
+            { community: { name: { _eq: $communityName } } }
+            {
+              _or: [
+                { player1: { name: { _eq: $playerName } } }
+                { player2: { name: { _eq: $playerName } } }
+              ]
+            }
+          ]
+        }
+        order_by: { date: desc }
+      ) {
+        ...ResultsTable_Results
+        edges {
+          node {
+            id
+            player1 {
+              id
+              name
+            }
+            player1goals
+            player2 {
+              id
+              name
+            }
+            player2goals
+            date
+            extratime
+          }
+        }
+      }
+    
+      community_settings_connection(
+        where: { community: { name: { _eq: $communityName } } }
+      ) {
+        edges {
+          node {
+            ...ExtraTimeColumn_IncludeExtraTime
+            ...ResultsTableHeader_CommunitySettings
+            score_type
+          }
+        }
+      }
+    }
+  |}
+];
+
+let toMatchResult =
+    (
+      resultNode: PlayerResultsQuery_graphql.Types.response_results_connection_edges_node,
+    ) => {
+  id: resultNode.id,
+  player1: {
+    id: resultNode.player1.id,
+    name: resultNode.player1.name,
+  },
+  player2: {
+    id: resultNode.player2.id,
+    name: resultNode.player2.name,
+  },
+  player1goals: resultNode.player1goals,
+  player2goals: resultNode.player2goals,
+  date: resultNode.date,
+  extratime: resultNode.extratime,
+};
 
 [@react.component]
 let make = (~playerName: string, ~communityName: string) => {
-  let (playerResultsQuery, _) =
-    useQuery(
-      ~variables=
-        PlayerResultsQuery.makeVariables(~communityName, ~playerName, ()),
-      PlayerResultsQuery.definition,
-    );
+  let queryData = Query.use(~variables={communityName, playerName}, ());
 
-  let settingsQuery = useCommunitySettings(communityName);
+  let resultsTableFragment = queryData.results_connection.fragmentRefs;
 
-  <>
-    <Header page={PlayerHome(communityName, playerName)} />
-    {switch (playerResultsQuery, settingsQuery) {
-     | (Loading, _)
-     | (_, Loading) => <MaterialUi.CircularProgress />
-     | (NoData, _)
-     | (_, NoData)
-     | (Error(_), _)
-     | (_, Error(_)) => <span> {text("Error")} </span>
-     | (Data(resultsData), Data(communitySettings)) =>
-       let results = resultsData##results |> toListOfResults;
-       let texts = Texts.getScoreTypeTexts(communitySettings.scoreType);
+  let communitySettings =
+    queryData.community_settings_connection.edges->Belt.Array.getExn(0);
```

Comment:
Is `getExn` going to crash when opening this page for newly created communities?

Reply:
Fixed in https://github.com/Yakimych/relogify/pull/103/commits/fcb61cb5a6d464d5a1efa2632f6a02af64bfe501